src/boot_x86_fsp.c

```diff
@@ -340,6 +398,33 @@ static int fsp_silicon_init(struct fsp_info_header *fsp_info, uint8_t *fsp_s_bas
     return EFI_SUCCESS;
 }
 
+#if defined(TARGET_x86_fsp_qemu) && defined(WOLFBOOT_MEASURED_BOOT)
+/* self-hash stage1/keystore/reset_vector. Assumes the flash layout described in
+ * hal/x86_fsp_qemu_stage1.ld.in */
+static int self_extend_pcr(void)
+{
+    uint8_t hash[SHA256_DIGEST_SIZE];
+    uint32_t blksz, position = 0;
+    wc_Sha256 sha256_ctx;
+    uint32_t sz;
+    uintptr_t p;
+
+    p  = (uintptr_t)_start_keystore;
+    sz = ((MEMORY_4GB) - (uint64_t)p);
```

Comment:
Is this `MEMORY_4GB` a physical limit for determining end of RAM? Would be nice to document or provide a helper for uses around end of memory location.